tests/test_weather_normalization.py

```diff
@@ -97,73 +86,53 @@ def test_weather_retrieval(results_dir, filename):
-    # They will be included in the tests again once simplified calibration is added.
-    # See https://github.com/NREL/OpenStudio-HPXML-Calibration/issues/43
-    if filename.name in SKIP_FILENAMES:
-        pytest.skip(f"Skipping test for {filename.name}")
-
-    hpxml = HpxmlDoc(filename)
-    inv_model = InverseModel(hpxml, user_config=test_config)
-    successful_fits = 0  # Track number of successful fits
-    output_filepath = results_dir / "weather_normalization"
-
-    plot_fuel_type_curve_fits(inv_model, output_filepath, filename.stem)
-
-    for fuel_type, bills in inv_model.bills_by_fuel_type.items():
-        model = inv_model.get_model(fuel_type)
-        bills_temps = inv_model.bills_weather_by_fuel_type_in_btu[fuel_type]
-        cvrmse = model.calc_cvrmse(bills_temps)
-
-        # Save CVRMSE result per test
-        individual_result = {f"{filename.stem}_{fuel_type}": cvrmse}
-        json_path = output_filepath / f"{filename.stem}_{fuel_type.value}.json"
-        with open(json_path, "w") as f:
-            json.dump(individual_result, f, indent=2)
-
-        successful_fits += 1
-
-    assert successful_fits > 0, (
-        f"No successful regression fits for {filename.stem}_{fuel_type.value}"
-    )
-
-
-# Use flaky to address intermittent tcl errors on Windows https://stackoverflow.com/questions/71443540/intermittent-pytest-failures-complaining-about-missing-tcl-files-even-though-the
-@pytest.mark.flaky(max_runs=3)
-# Suppress warning ("invalid value encountered in sqrt") that is only emitted during testing, not during normal use
-# See discussion in https://github.com/NREL/OpenStudio-HPXML-Calibration/issues/75
-@pytest.mark.filterwarnings("ignore:invalid value encountered in sqrt:RuntimeWarning")
-# Skipping because of this bug in Python https://github.com/python/cpython/issues/125235#issuecomment-2412948604
-@pytest.mark.skipif(
-    sys.platform == "win32"
-    and sys.version_info.major == 3
-    and sys.version_info.minor == 13
-    and sys.version_info.micro <= 2,
-    reason="Skipping Windows and Python <= 3.13.2 due to known bug",
-)
-@pytest.mark.parametrize(
-    "filename", ira_rebate_hpxmls + real_home_hpxmls + ihmh_home_hpxmls, ids=lambda x: x.stem
-)
-def test_fit_model(filename):
-    if filename.name in SKIP_FILENAMES:
-        pytest.skip(f"Skipping test for {filename.name}")
-    hpxml = HpxmlDoc(filename)
-    inv_model = InverseModel(hpxml, user_config=test_config)
-    fuel_types = hpxml.get_fuel_types()
-    conditioning_fuels = fuel_types["heating"] | fuel_types["cooling"]
-
-    for fuel_type, bills in inv_model.bills_by_fuel_type.items():
-        bills_temps = inv_model.bills_weather_by_fuel_type_in_btu[fuel_type]
-
-        best_fitting_model = fit_model(
-            bills_temps,
-            test_config["acceptance_criteria"]["bill_regression_max_cvrmse"],
-            conditioning_fuels,
-            fuel_type,
+def test_curve_fit_and_fit_model(results_dir, filename):
+    try:
```

Comment:
It looks like a lot of code change, but really I just wrapped the code in try/except, and ensure that we hit the exception as expected for certain test files.